crates/ruma-events/src/content.rs

```diff
@@ -30,6 +30,15 @@ pub trait StaticEventContent: Sized {
     const TYPE: &'static str;
 }
 
+/// An event content type with a dynamic event `type` value.
+///
+/// A dynamic event type has a statically-known type prefix, and the rest of the type is stored in a
+/// field in its content type.
+pub trait DynamicEventContent: Sized {
+    /// The prefix of the event type.
+    const TYPE_PREFIX: &'static str;
```

Comment:
Maybe note that this would usually / always end in `.` (i.e. the first half of the `.*` is included in this)?

Reply:
Would this be okay:

> This should usually end with a `.`, which is used as a separator in Matrix event types.

?

Reply:
Yup.